docs/DeveloperGuide.md

```diff
@@ -4,35 +4,261 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
+### Architecture Diagram
+![Architecture Diagram](images/ArchitectureDiagram.png)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+*Figure 1: Architecture Diagram*
 
+The Architecture Diagram given above explains the high-level design of the App.
+Listed below is a brief summary outlining the primary components and their interrelationships.
 
-## Product scope
-### Target user profile
+In summary, the user interacts with the UI components, initiating a sequence that involves:
+- Parser component for command interpretation
+- Command component for execution 
+- Data component for managing application data
 
-{Describe the target user profile}
+The Storage component mainly handles interaction with external text files and main coordinates the interactions between the various Components.
 
-### Value proposition
+The bulk of the app’s work is done by the following components:
+- `UI` : The UI of the App.
+- `Storage` : Reads data from, and writes data to, the text files.
+- `Data` : Consists of all the classes that are involved in execution of commands.
+- `Parser` : Makes sense of user input to return the appropriate command
+- `Command` : Executes the comman requested by the user.
+
+How the architecture components interact with each other
```

Comment:
Should this be a heading?

Looking at it on the website, it seems like normal text; but I think it signifies a new segment. See this: 
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/49584216/7ff5f916-af8e-441e-b035-74dc102c0b12)

Reply:
Thanks for pointing out! I think your suggestion makes sense and I will edit accordingly!